cluster/cluster.go

```diff
@@ -110,12 +115,18 @@ func (c *cluster) Unregister(_ context.Context, req *clusterpb.UnregisterRequest
 		}
 	}
 	if index < 0 {
-		return nil, fmt.Errorf("address %s has  notregistered", req.ServiceAddr)
+		return nil, fmt.Errorf("address %s has not registered", req.ServiceAddr)
 	}
-
+	// as UnregisterCallback arg
+	unregisterMember := *c.members[index]
```

Comment:
Seems we don't need this variable.
```go
	if c.currentNode.UnregisterCallback != nil {
		c.currentNode.UnregisterCallback(*c.members[index])
	}
```

Reply:
Yes, since there is no possibility of exception, I can advance the method to delMeber